contracts/RelayerHub.sol

```diff
@@ -8,88 +8,149 @@ import "./System.sol";
+    }
+
+    function whitelistInit() external {
+        require(!whitelistInitDone, "the whitelists already updated");
+        addInitRelayer(WHITELIST_1);
+        addInitRelayer(WHITELIST_2);
+        whitelistInitDone = true;
+    }
+
+    function addInitRelayer(address addr) internal {
+        relayManagersExistMap[addr] = true;
+        managerToRelayer[addr] = addr; // for the current whitelisted relayers we are keeping manager and relayer address the same
+        currentRelayers[addr] = true;
+    }
+
+    /*********************** Param update ********************************/
+    function updateParam(string calldata key, bytes calldata value) external override onlyInit onlyGov {
+        if (Memory.compareStrings(key, "addManager")) {
+            require(value.length == 20, "length of manager address mismatch");
+            address newManager = BytesToTypes.bytesToAddress(20, value);
+            addManagerByGov(newManager);
+        } else if (Memory.compareStrings(key, "removeManager")) {
+            require(value.length == 20, "length of manager address mismatch");
+            address payable managerAddress = payable(BytesToTypes.bytesToAddress(20, value));
+            removeManager(managerAddress);
+        } else {
+            require(false, "unknown param");
+        }
+        emit paramChange(key, value);
+    }
+
+    function removeManagerByHimself() external {
+        // here the manager removes himself
+        removeManager(payable(msg.sender));
+    }
+
+    function removeManager(address payable managerAddress) internal {
+        // check if the manager address already exists
+        require(relayManagersExistMap[managerAddress], "manager doesn't exist");
+
+        address relayerAddress = managerToRelayer[managerAddress];
+
+        delete (relayManagersExistMap[managerAddress]);
+        delete (managerToRelayer[managerAddress]);
+
+        // emit success event
+        emit managerRemoved(managerAddress);
+        if (relayerAddress != address(0)) {
+            delete (currentRelayers[relayerAddress]);
+            emit relayerUpdated(relayerAddress, address(0));
+        }
+    }
+
+    function addManagerByGov(address managerToBeAdded) internal {
+        require(!relayManagersExistMap[managerToBeAdded], "manager already exists");
+        require(!isContract(managerToBeAdded), "contract is not allowed to be a manager");
+
+        relayManagersExistMap[managerToBeAdded] = true;
+
+        emit managerAdded(managerToBeAdded);
+    }
+
+    // updateRelayer() can be used to add relayer for the first time, update it in future and remove it
+    // in case of removal we can simply update it to a non-existing account
+    function updateRelayer(address relayerToBeAdded) public onlyManager {
+        require(!isContract(relayerToBeAdded), "contract is not allowed to be a relayer");
+
+        address oldRelayer = managerToRelayer[msg.sender];
+        delete currentRelayers[oldRelayer];
```

Comment:
This line should be moved to `line 145`. While `relayerToBeAdded == oldRelayer`, it will not fail to update as expected since `line 136`